### DISCUSSION TITLE
BattyBirdNET-Pi  on Pi Zero W 2

### DISCUSSION THREAD

**Discussion by Inachis** [rdz-oss/BattyBirdNET-Pi#14] [Q&A]
I finally got around to trying out a Zero W 2 powered with Anker 20000mAh 525 Power Bank.

Initially I installed BirdNET-Pi following the usual installation instructions modified as at https://github.com/mcguirepr89/BirdNET-Pi/wiki/RPi0W2-Installation-Guide. Got just under 24 hours from the power pack with similar results to the Pi 4B, not quite as good but it was using a different type of microphone (a spare PS3 eye USB Webcam!).

Now moved on to BattyBirdNET-Pi with the AudioMoth configured as USB microphone. Followed same configuration but using the BattyBirdNET-Pi script. For BirdNET-Pi it was recommended to to adjust the "Recording Length" up to 30 seconds (from default 15 seconds) in Advanced Settings. But the BattyBirdNET-Pi default Recording Length is 3 seconds...and BattyBirdNET-Pi analysis is now way way behind, about 35 mins after running for 45 minutes...like when I had the AudioMoth set to 384kHz and it was running too slow.

So ideas on configuration for Zero W 2?

**Comment by rdz-oss** [comment]
I have to check the hardware specs of the Zero. Maybe it is not powerful enough. Some info on that:
- The system samples the input always to 256kHZ. So it overloads it if you use 384kHz. Unnecesary load that has no benefit detection wise. Consider using 250kHz .
- the system scans at 256kHz rather than 48kHz as compared for birds. So all operations are on data that is 5 times larger than for birdnet.
- The system has to scan at 2 times per second rather than once per 3 seconds on the bird version (the chuncks on why ids run, not recording length). So it does 6 id operations for bats where birdnet would do 1

--> So the computational effort is significantly higher for bats than for birds.

--> The memory used on my Pi B is 2GB . Would also need to cross check with the Zero.

Perhaps the system can be made more efficient.  Let me check the Zero specs.

BR
Richard
